swim/memberlist.go

```diff
@@ -225,6 +230,31 @@ func (m *memberlist) MakeLeave(address string, incarnation int64) []Change {
 	return m.MakeChange(address, incarnation, Leave)
 }
 
+func (m *memberlist) MakeTombstone(address string, incarnation int64) []Change {
+	m.node.emit(MakeNodeStatusEvent{Tombstone})
+	return m.MakeChange(address, incarnation, Tombstone)
+}
+
+func (m *memberlist) Evict(address string) {
+	if m.local.Address == address {
+		// don't evict ourself, this is an assertion that no execution
+		// path should lead to this state
+		panic("can't evict self")
+	}
+	m.members.Lock()
+	if member, ok := m.members.byAddress[address]; ok {
+		delete(m.members.byAddress, address)
+		for i, lMember := range m.members.list {
+			if member == lMember {
+				// a safe way to remove a pointer from a slice
+				m.members.list, m.members.list[len(m.members.list)-1] = append(m.members.list[:i], m.members.list[i+1:]...), nil
```

Comment:
Can we abstract memberlist manipulation to a separate module, or just a list of functions? Because memberlist update code is becoming sprinkled throughout this module...